### DISCUSSION TITLE
Customization features added

### DISCUSSION THREAD

**Discussion by Bluecatzmv** [netbox-community/netbox#12385] [Ideas]
### NetBox version

v3.1.7

### Feature type

Change to existing functionality

### Proposed functionality

The existing user customization feature is not flexible enough, the custom fields are not retrievable by the system, they are simply fields, and there is no option to set the custom fields as new information that can be jumped to edit. The custom link option does not point to a small enough scope of the directory, it will display all the sub items under the major category of the directory, the user cannot choose to have the custom link display in the smallest category of the individual information pages.

### Use case

For example, after creating a server device, adding device properties to it, where information such as hard disk, memory stick, CPU, network card, etc. can only be added by customization, but the custom fields cannot be retrieved by the system, and the custom fields are simply in the form of strings, and cannot be bound to the device, or jumped by clicking on the custom fields in the device interface, because currently it is not possible to bind and jump between the device and the device, while for devices such as servers, the existing information options cannot meet the demand, and a lot of information needs to be added by customization.

The flexibility of custom links is also not enough, and the subdivision of the specified type is not fine enough, if it can be subdivided into each minimum page, it will greatly improve the experience of using.

**Comment by jeremystretch** [comment]
Converting this to a discussion as it's not clear what, if any, changes are actually being proposed.

**Comment by candlerb** [comment]
Try an up-to-date version of Netbox first, e.g. the online version at demo.netbox.dev (EDIT: currently out-of-service)

* [v3.2.0](https://github.com/netbox-community/netbox/releases/v3.2.0) introduced "object" as a custom field type, so a custom field can reference another object within Netbox in a foreign-key-like way
* [v3.3.0](https://github.com/netbox-community/netbox/releases/v3.3.0) allows custom fields to be grouped, and visibility toggled
* [v3.4.0](https://github.com/netbox-community/netbox/releases/v3.4.0) includes custom fields in global search results
* [v3.5.0](https://github.com/netbox-community/netbox/releases/v3.5.0) preserves custom fields when cloning an object

I think you'll find custom fields are more usable as a result of these changes.

v3.5.0 is brand new, so I'd suggest you upgrade to v3.4.10 for now.

Once you've done this, you can make more specific suggestions, with specific examples of where you're using custom fields today, what the workflow is today, and what you'd like the workflow to become.
